sql/src/test/java/io/crate/integrationtests/SeqNoBasedOCCIntegrationTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.integrationtests;
+
+import io.crate.action.sql.SQLActionException;
+import io.crate.exceptions.VersioninigValidationException;
+import org.junit.Test;
+
+import static io.crate.testing.TestingHelpers.printedTable;
+import static org.hamcrest.core.Is.is;
+
+public class SeqNoBasedOCCIntegrationTest extends SQLTransportIntegrationTest {
+
+    @Test
+    public void testDeleteWhereSeqNoAndTerm() throws Exception {
+        execute("create table t (x integer primary key, y string) with (number_of_replicas=0)");
+        ensureYellow();
+
+        execute("insert into t (x, y) values (?, ?)", new Object[]{1, "don't panic"});
+        refresh();
+
+        execute("select _seq_no, _primary_term from t where x = 1");
+        assertEquals(1L, response.rowCount());
+        Long seqNo = (Long) response.rows()[0][0];
+        Long primaryTerm = (Long) response.rows()[0][1];
+
+        execute("delete from t where x = 1 and _seq_no = ? and _primary_term = ?",
+            new Object[]{seqNo, primaryTerm});
+        assertEquals(1L, response.rowCount());
+
+        // Validate that the row is really deleted
+        refresh();
+        execute("select * from t where x = 1");
+        assertEquals(0, response.rowCount());
```

Comment:
Could move the comment above into the assertion as message:
```suggestion
        assertThat("Records with x=1 must be deleted", response.rowCount(), is(0L));
```